drivers/wifi/nrf700x/src/fw_load.c

```diff
@@ -63,23 +64,28 @@ LOG_MODULE_DECLARE(wifi_nrf, CONFIG_WIFI_NRF700X_LOG_LEVEL);
 	extern                  const char prefix ## _ ## name ## _end[];
 
 INCBIN(_bin, nrf70_fw, STR(CONFIG_NRF_WIFI_FW_BIN));
+#endif /* CONFIG_NRF_WIFI_PATCHES_BUILTIN */
 
 enum nrf_wifi_status nrf_wifi_fw_load(void *rpu_ctx)
 {
 	enum nrf_wifi_status status = NRF_WIFI_STATUS_FAIL;
 	struct nrf_wifi_fmac_fw_info fw_info = { 0 };
-#if defined(CONFIG_NRF_WIFI_PATCHES_EXT_FLASH) && defined(CONFIG_NORDIC_QSPI_NOR)
+#if defined(CONFIG_NRF_WIFI_PATCHES_EXT_FLASH_XIP) && defined(CONFIG_NORDIC_QSPI_NOR)
 	const struct device *flash_dev = DEVICE_DT_GET(DT_INST(0, nordic_qspi_nor));
-#endif /* CONFIG_NRF_WIFI_PATCHES_EXT_FLASH */
+#endif /* CONFIG_NRF_WIFI_PATCHES_EXT_FLASH_XIP */
 
+#ifdef CONFIG_NRF_WIFI_PATCHES_BUILTIN
 	status = nrf_wifi_fmac_fw_parse(rpu_ctx, (const uint8_t *)_bin_nrf70_fw_start,
 					_bin_nrf70_fw_end - _bin_nrf70_fw_start, &fw_info);
 	if (status != NRF_WIFI_STATUS_SUCCESS) {
 		LOG_ERR("%s: nrf_wifi_fmac_fw_parse failed", __func__);
 		return status;
 	}
+#else
+	BUILD_ASSERT(0, "CONFIG_NRF_WIFI_PATCHES_BUILTIN must be enabled");
```

Comment:
So this option is not really useful until non-XIP patch support is added. Is that going to happen in NCS 2.6.0?

Reply:
No, it works even now, but for this option we also need `CONFIG_NRF_WIFI_PATCHES_BUILTIN` as that builds the patch in to the image and then XIP relocates.

Reply:
> Is that going to happen in NCS 2.6.0?

Yes, this patch is in preparation for that. But it's limited to non-XIP over QSPI i.e., 5340/52840.